### PR TITLE
docs(carousel): refactor carousel stories to use csf and mdx - FE-3442

### PR DESCRIPTION
**Pull request by mkrds**
This PR refactors `Carousel` stories to remove the deprecated `storiesOf` and use `csf` / `mdx` formats instead.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

**Review by grabkowski — DISMISSED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

File: cypress/features/regression/carousel.feature [hunk 1]
```diff
@@ -1,14 +1,9 @@
 Feature: Carousel component
   I want to check Carousel properties
 
-  @positive
-  Scenario: Initial slide title is Slide 3
```
Comment (KatarzynaQA) [inline cypress/features/regression/carousel.feature:5]: why did you remove it ? I don't see it's covered by chromatic

Comment (DlgSHi) [reply cypress/features/regression/carousel.feature:5]: here is an example :) https://www.chromatic.com/test?appId=5ecf782fe724630022d27d7d&id=5fdb5c4c0d86c5002104c5b9

**Review by KatarzynaQA — APPROVED** [review]

**Review by Alec-Walker-Sage — DISMISSED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by Alec-Walker-Sage — APPROVED** [review]

**Review by nicktitchmarsh — APPROVED** [review]
